modules/nextflow/src/main/groovy/nextflow/trace/TraceObserver.groovy

```diff
@@ -123,11 +123,18 @@ trait TraceObserver {
     void onFlowError(TaskHandler handler, TraceRecord trace){}
 
     /**
-     * Method that is invoked when a value is published from a channel.
+     * Method that is invoked when a workflow output is published.
      *
+     * @param name
+<<<<<<< HEAD
+     *      The name of the workflow output
+=======
+>>>>>>> 3289e9e30 (first POC output structure and annotations)
      * @param value
+     *      A list if the published channel was a queue channel,
+     *      otherwise an object if the channel was a value channel
      */
-    void onWorkflowPublish(Object value){}
+    void onWorkflowPublish(String name, Object value){}
```

Comment:
Is this the workflow run name? is it really necessary ?

Reply:
It is the name of the workflow output, of which there can be several

Reply:
Don't think that is needed because it is an undocumented preview feature (part of workflow outputs preview)

Reply:
It is the name given to the output in the workflow. We use it as key for the outputs map, and it is also used to refer to an specific output by cid://<hash>/#output.<name>

Reply:
Regarding the API change, let's move the discussion to #5909 which is the source of truth for this change

Reply:
> Don't think that is needed because it is an undocumented preview feature

Fair enough